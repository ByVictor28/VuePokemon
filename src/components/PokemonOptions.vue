<template>
  <div class="flex flex-col items-center">
    <ul class="">
      <li 
        class="bg-red-200 my-4 w-36 py-1 hover:bg-red-400 rounded-md cursor-pointer transition-colors"
        v-for="pokemon in pokemons"
        :key="pokemon.id"
        @click=" $emit('selection',pokemon.id)"
      >
        <span :class="[showResult && pokemon.correct?'text-green-400':'text-red-400']">{{pokemon.name}}</span>
      </li>
    </ul>
    
    
  </div>
</template>

<script>
export default {
  props:{
    pokemons:{
      required:true
    },
    showResult:{
      
    }
  },
  watch:{
    pokemons:(value)=>{
      console.log(value)
    }
  }
}
</script>

<style>

</style>