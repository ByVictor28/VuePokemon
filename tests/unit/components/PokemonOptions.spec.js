import {shallowMount} from "@vue/test-utils"
import PokemonOptions from "@/components/PokemonOptions"
import { pokemons } from "../mocks/pokemons.mock"


describe('PokemonOptions component', () => {
  test('sreenshot match', () => {
    const wrapper = shallowMount(PokemonOptions,{
      props:{
        pokemons
      }
    }) 
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('show the 4 options correctly', () => {
    const wrapper = shallowMount(PokemonOptions,{
      props:{
        pokemons
      }
    }) 
    const [li1,li2,li3,li4] = wrapper.findAll("li")
    
    expect(li1.text()).toBe(pokemons[0].name)
    expect(li2.text()).toBe(pokemons[1].name)
    expect(li3.text()).toBe(pokemons[2].name)
    expect(li4.text()).toBe(pokemons[3].name)
  })

  test('emit selection event when clicking a li', () => {
    const wrapper = shallowMount(PokemonOptions,{
      props:{
        pokemons
      }
    }) 
    const [li1,li2,li3,li4] = wrapper.findAll("li")

    li1.trigger("click")
    li2.trigger("click")
    li3.trigger("click")
    li4.trigger("click")

    expect(wrapper.emitted("selection")[0]).toEqual([pokemons[0].id])
    expect(wrapper.emitted("selection")[1]).toEqual([pokemons[1].id])
    expect(wrapper.emitted("selection")[2]).toEqual([pokemons[2].id])
    expect(wrapper.emitted("selection")[3]).toEqual([pokemons[3].id])
  })
  
  
  
})
