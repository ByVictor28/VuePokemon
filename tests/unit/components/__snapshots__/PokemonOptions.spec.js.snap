// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component sreenshot match 1`] = `
<div class="flex flex-col items-center">
  <ul class="">
    <li class="bg-red-200 my-4 w-36 py-1 hover:bg-red-400 rounded-md cursor-pointer transition-colors"><span class="text-red-400">tynamo</span></li>
    <li class="bg-red-200 my-4 w-36 py-1 hover:bg-red-400 rounded-md cursor-pointer transition-colors"><span class="text-red-400">mr-mime</span></li>
    <li class="bg-red-200 my-4 w-36 py-1 hover:bg-red-400 rounded-md cursor-pointer transition-colors"><span class="text-red-400">mandibuzz</span></li>
    <li class="bg-red-200 my-4 w-36 py-1 hover:bg-red-400 rounded-md cursor-pointer transition-colors"><span class="text-red-400">spheal</span></li>
  </ul>
</div>
`;
